app/models/mhv_account.rb

```diff
@@ -109,29 +112,41 @@ def create_mhv_account!
     if may_register?
       client_response = mhv_ac_client.post_register(params_for_registration)
       if client_response[:api_completion_status] == 'Successful'
+        StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.success")
+
         user.va_profile.mhv_ids = [client_response[:correlation_id].to_s]
         user.instance_variable_get(:@mvi).save
         self.registered_at = Time.current
         register!
       end
       # TODO: be prepared to handle or raise various exceptions
     end
+  ensure
+    StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.total") if may_register? || registered?
   end
 
   def upgrade_mhv_account!
-    if may_upgrade?
+    initially_premium = !may_upgrade?
```

Comment:
The problem with that would be that may_upgrade will return false in the ensure block after the call to MHV, and so we won't know in the ensure block if the call to MHV was made. We need to differentiate between accounts that were already premium and not calling MHV and accounts we aren't sure are premium and calling mhv. By setting initially premium to may_upgrade in the beginning